src/poetry/utils/_compat.py

```diff
@@ -50,10 +55,18 @@ def encode(string: str, encodings: list[str] | None = None) -> bytes:
     return string.encode(encodings[0], errors="ignore")
 
 
+def is_relative_to(this: Path, other: Path) -> bool:
```

Comment:
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to

I had to think twice, too. A comment makes sense.

Reply:
Moved it to helpers as I just noticed this nugget in the doc.

> Deprecated since version 3.12, will be removed in version 3.14: Passing additional arguments is deprecated; if supplied, they are joined with other.

Reply:
For me, this is somewhat ambiguous. I'm wondering if the whole method is deprecated or only passing additional arguments?

Reply:
https://github.com/python/cpython/pull/94469/files#diff-1134e36a94ecfde1df43bee5efd285de5d67426fbca086425201ecc753f9139cR612-R615

You might be right @radoering. This could be clearer. But I think the PR makes the intention clear. I will rework accordingly.